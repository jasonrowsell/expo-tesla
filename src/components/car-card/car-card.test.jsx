import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import CarCard from './car-card';

describe('<CarCard />', () => {
  const renderComponent = () => render(<CarCard title="title" tagline="tagline" source="img" />);
  it('has 2 children', () => {
    const tree = renderer.create(<CarCard title="title" tagline="tagline" source="img" />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it('renders correctly', () => {
    const tree = renderComponent();
    expect(tree).toMatchSnapshot();
  });

  it('has the title', () => {
    const { getByText } = renderComponent();
    getByText('title');
  });

  it('has the tagline', () => {
    const { getByText } = renderComponent();
    getByText('tagline');
  });
});
