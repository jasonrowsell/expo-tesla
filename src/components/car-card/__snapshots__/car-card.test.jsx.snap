// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarCard /> renders correctly 1`] = `
<View
  style={
    Object {
      "height": 1334,
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={null}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": "30%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "500",
        }
      }
    >
      title
    </Text>
    <Text
      style={
        Object {
          "color": "#5c5e62",
          "fontSize": 16,
        }
      }
    >
      tagline
    </Text>
  </View>
</View>
`;
